### PR TITLE
chore: fix surge preview workflow

### PR DESCRIPTION
**Pull request by afc163**

### CODE REVIEW

File: .github/workflows/preview-build.yml [hunk 1]
```diff
@@ -1,60 +1,41 @@
...
-      - run: |
-          zip -r dist.zip dist
-
-      - name: Upload dist artifact
+      - uses: actions/checkout@v4
+      - uses: oven-sh/setup-bun@v2
+      - run: bun install
+      - name: bun run site
+        id: site
+        run: bun run site
```
Comment (afc163) [inline .github/workflows/preview-build.yml:27]:
```suggestion
        run: bun run build-doc
```

File: .github/workflows/preview-build.yml [hunk 2]
```diff
@@ -1,60 +1,41 @@
...
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+      - name: run e2e test
+        run: bun run test:site
```
Comment (afc163) [inline .github/workflows/preview-build.yml:31]:
```suggestion
```

File: .github/workflows/preview-build.yml [hunk 3]
```diff
@@ -1,60 +1,41 @@
...
+      - name: upload site artifact
         uses: actions/upload-artifact@v4
         with:
-          name: dist
-          path: dist.zip
+          name: site
+          path: _site/
```
Comment (afc163) [inline .github/workflows/preview-build.yml:36]:
```suggestion
          name: dist
          path: dist/
```

File: .github/workflows/preview-build.yml [hunk 4]
```diff
@@ -1,60 +1,41 @@
...
-          pnpm build-doc
-
-      - run: |
-          zip -r dist.zip dist
-
-      - name: Upload dist artifact
+      - uses: actions/checkout@v4
+      - uses: oven-sh/setup-bun@v2
+      - run: bun install
+      - name: bun run site
```
Comment (afc163) [inline .github/workflows/preview-build.yml:25]:
```suggestion
      - name: build site
```

File: .github/workflows/preview-deploy.yml [hunk 5]
```diff
@@ -6,72 +8,119 @@ on:
...
+        uses: dawidd6/action-download-artifact@v6
         with:
           workflow: ${{ github.event.workflow_run.workflow_id }}
-          workflow_conclusion: success
-          name: dist
-      - run: |
-          unzip dist.zip
-      - name: upload to surge
+          run_id: ${{ github.event.workflow_run.id }}
+          name: site
```
Comment (afc163) [inline .github/workflows/preview-deploy.yml:88]:
```suggestion
          name: dist
```

File: .github/workflows/preview-deploy.yml [hunk 6]
```diff
@@ -6,72 +8,119 @@ on:
...
+
+      - name: success comment
+        uses: actions-cool/maintain-one-comment@v3
+        if: ${{ steps.deploy.outcome == 'success' }}
         with:
-          issue-number: ${{ steps.pr.outputs.id }}
+          token: ${{ secrets.GITHUB_TOKEN }}
           body: |
-            Deploy PR preview failed.
+            [<img width="300" alt="Preview is ready" src="https://user-images.githubusercontent.com/5378891/72400743-23dbb200-3785-11ea-9d13-1a2d92743846.png">](https://preview-${{ steps.pr.outputs.id }}-ant-design.surge.sh)
```
Comment (afc163) [inline .github/workflows/preview-deploy.yml:105]:
```suggestion
            [<img width="300" alt="Preview is ready" src="https://user-images.githubusercontent.com/5378891/72400743-23dbb200-3785-11ea-9d13-1a2d92743846.png">](https://preview-${{ steps.pr.outputs.id }}-ant-design-mobile.surge.sh)
```

File: .github/workflows/preview-deploy.yml [hunk 7]
```diff
@@ -6,72 +8,119 @@ on:
...
-        id: pr
-        run: echo "::set-output name=id::$(<pr-id.txt)"
-      - name: The job failed
-        uses: peter-evans/create-or-update-comment@v2
-        with:
-          issue-number: ${{ steps.pr.outputs.id }}
+          token: ${{ secrets.GITHUB_TOKEN }}
           body: |
-            Deploy PR preview failed.
+            [<img width="300" alt="Preview failed" src="https://user-images.githubusercontent.com/5378891/75333447-1e63a280-58c1-11ea-975d-235367fd1522.png">](https://preview-${{ steps.pr.outputs.id }}-ant-design.surge.sh)
```
Comment (afc163) [inline .github/workflows/preview-deploy.yml:116]:
```suggestion
            [<img width="300" alt="Preview failed" src="https://user-images.githubusercontent.com/5378891/75333447-1e63a280-58c1-11ea-975d-235367fd1522.png">](https://preview-${{ steps.pr.outputs.id }}-ant-design-mobile.surge.sh)
```

File: .github/workflows/preview-start.yml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      pull-requests: write  # for actions-cool/maintain-one-comment to modify or create PR comments
+    name: start preview info
+    runs-on: ubuntu-latest
+    steps:
+      - name: update status comment
+        uses: actions-cool/maintain-one-comment@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body: |
+            [<img width="500" alt="Prepare preview" src="https://user-images.githubusercontent.com/5378891/72351368-2c979e00-371b-11ea-9652-eb4e825d745e.gif">](https://preview-${{ github.event.number }}-ant-design.surge.sh)
```
Comment (afc163) [inline .github/workflows/preview-start.yml:29]:
```suggestion
            [<img width="500" alt="Prepare preview" src="https://user-images.githubusercontent.com/5378891/72351368-2c979e00-371b-11ea-9652-eb4e825d745e.gif">](https://preview-${{ github.event.number }}-ant-design-mobile.surge.sh)
```

File: .github/workflows/preview-build.yml [hunk 9]
```diff
@@ -1,60 +1,39 @@
...
-
-      - name: Run Build Script
-        run: |
-          pnpm build-doc
-
-      - run: |
-          zip -r dist.zip dist
-
-      - name: Upload dist artifact
+      - uses: actions/checkout@v4
```
Comment (afc163) [inline .github/workflows/preview-build.yml:22]:
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 16


```

File: .github/workflows/preview-build.yml [hunk 10]
```diff
@@ -1,60 +1,43 @@
...
-            ${{ runner.os }}-pnpm-store-
-
-      - name: Install dependencies
-        run: pnpm install
-
-      - name: Run Build Script
-        run: |
-          pnpm build-doc
-
-      - run: |
```
Comment (afc163) [inline .github/workflows/preview-build.yml:48]: - run: |
          zip -r dist.zip dist

File: .github/workflows/preview-build.yml [hunk 11]
```diff
@@ -1,60 +1,43 @@
...
+        id: site
+        run: bun run build-doc
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+      - name: upload site artifact
         uses: actions/upload-artifact@v4
         with:
           name: dist
-          path: dist.zip
+          path: dist/
```
Comment (afc163) [inline .github/workflows/preview-build.yml:38]:
```suggestion
          path: dist.zip
```

**Review by zombieJ — APPROVED** [review]

**Pull request merged**